src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -79,4 +83,11 @@ public void deleteBook(@PathVariable Long id, @PathVariable Long idBook) {
                 .removeBook(book);
     }
 
+    @RequestMapping(value = "/logueduser", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public String currentUser(Principal principal) {
```

Comment:
This works great.
FYI, this is another way of doing this without injecting the Principal in a method:

    private OrangeUser getCurrentUser() {
        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
        if (auth != null && !auth.getPrincipal().equals("anonymousUser")) {
            if (auth.getPrincipal() instanceof String) {
                return userRepository.findOneByUsername((String) auth.getPrincipal());
            } else {
                User user = (User) auth.getPrincipal();
                return userRepository.findOneByUsername(user.getUsername());
            }
        }

        return null;
    }